fpgalib/docs/GHzDAC_v8.md

```diff
@@ -0,0 +1,505 @@
+The data consists of 16 bytes of CountTo counter limits and 64*8 bytes of Jumptable entries.
+
+fromAddr and toAddr are addresses of 4ns blocks. If we set StartAdr `d(17..19)` to 0 and a jump table END with FromAdr=10, SRAM will run for 10*4ns=40ns before hitting the END command.
+
+    l(1)  length[15..8]   set to 2  (528 = 16+64*8)
+    l(2)  length[7..0]    set to 16
+
+    d(1)  CountTo0[7..0]  CountTo0[31..0] is maximum count of counter 0
+    d(2)  CountTo0[15..8]
+    d(3)  CountTo0[23..16]
+    d(4)  CountTo0[31..24]
+
+    d(5)  CountTo1[7..0]  CountTo1[31..0] is for counter 1
+    ...
+    d(16) CountTo3[7..0]  CountTo3[31..0] is for counter 3
+
+    d(17) StartAdr[7..0]    3 bytes for StartAdr[23..0]
+    d(18) StartAdr[15..8]
+    d(19) StartAdr[23..16]
+    d(20) StartAdr[7..0]    Must be equal to d(17),d(18),d(19)
+    d(21) StartAdr[15..8]
+    d(22) StartAdr[23..16]
+    d(23) StartOpCode[7..0]  =5 for NOP
+    d(24) StartOpCode[15..8] =0
+
+    d(25) FromAdr(1)[7..0]  3 bytes for FromAdr(1)[23..0]-1, jumptable index=1
+    d(26) FromAdr(1)[15..8]
+    d(27) FromAdr(1)[23.16]
+    d(28) ToAdr(1)[7..0]    3 bytes for ToAdr(1)[23..0]
+    d(29) ToAdr(1)[15..8]
+    d(30) ToAdr(1)[23.16]
+    d(32) OpCode(1)[7..0]   2 bytes for OpCode(1)[15..0]
+    d(32) OpCode(1)[15..8]
+
+    d(33) FromAdr(2)[7..0]  Bytes for FromAdr(2)[23..0]-1, index=2
+    ...
+    d(528) OpCode(63)[15..8] Bytes for OpCode(63)[15..8], index=63
+
+
+### Operation codes for JumpTable:
+
+    dddddddd ddddddd0   IDLE: extra d[14..0] 4ns clocks (max 128 us)
+                          (2**15)-1 *4ns = 131068ns
+                          Does this mean minimum idle time is 2*4ns = 8ns?
+                          Is the lowest order bit ignored or what? Maybe lowest bit=0 indicates IDLE.
+
+    xxjjjjjj iiiin001   CHECK: if daisychain bit(index=iiii) = n
+                          T:          jump ToAdr, load JumpIndex jjjjjj
+                          F:          AdrCounter+1,    JumpIndex+1
+
+    xxjjjjjj xxxx1101   JUMP:         jump ToAdr, load JumpIndex jjjjjj
+
+    xxxxxxxx xxxx0101   NOP:          AdrCounter+1,    JumpIndex+1
+
+    xxjjjjjj xxccx011   CYCLE: until count(ind=cc) = CountTo(ind=cc)
+                          F: count+1, jump ToAdr, load JumpIndex jjjjjj
+                          T: count=0, AdrCounter+1,    JumpIndex+1
+
+    xxxxxxxx xxxxx111   END: SRAMend, stops Adr counting at FromAdr+2
+
+    x=do not care.
+
+JumpTable addresses FromAdr must be 1 less than actual address (due to pipelining).
+For StartAdr (index=0 of JumpTable), set FromAdr and ToAdr to actual address.
+For END, set FromAdr to 2 less than actual stop address, where Adr will idles.
+ToAdr is actual value.
+
+`CHECK` evaluated at address FromAdr+1
+
+`IDLE` produces d+1 clock cycles at address FromAdr+1. (ToAdr not used in IDLE) setting to 0 (`dd...dd=0`) produces 1 clock cycle of delay for the instruction itself.
```

Comment:
The punctuation in this paragraph is a bit off in a few places. Give it a once-over.